app/controllers/application_controller.rb

```diff
@@ -2,7 +2,8 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
-
+  before_action :authenticate_user!, :except => [:index]
```

Comment:
This is a great tip, thank you Eliot!!

On Fri, Oct 16, 2015 at 10:58 AM, Eliot Sykes <notifications@github.com>
wrote:

> In app/controllers/application_controller.rb
> <https://github.com/HelenaWu/blocitoff/pull/5#discussion_r42251981>:
>
> > @@ -2,7 +2,8 @@ class ApplicationController < ActionController::Base
> >    # Prevent CSRF attacks by raising an exception.
> >    # For APIs, you may want to use :null_session instead.
> >    protect_from_forgery with: :exception
> > -
> > +  before_action :authenticate_user!, :except => [:index]
>
> This is a secure choice, having the authenticate_user! method called
> before *almost* every action on every controller that inherits from
> ApplicationController.
>
> However, its configured to not be applied to any index actions, which I
> think could be easily forgotten when creating any new controllers.
>
> Supposing a new controller was added to the application, that showed
> sensitive/private data as part of its index action, we'd need to remember
> to re-configure the authenticate_user! call in the new controller to
> protect the index action.
>
> This isn't ideal as forgetfulness could lead to private data being exposed
> accidentally. Its usually better to make forgetfulness cause an application
> to break in a secure way that will over-protect data rather than
> under-protect data.
>
> A slight improvement on the above would be to change ApplicationController
> to configure this as:
>
>    before_action :authenticate_user!
>
> and in any controllers that need actions to *not* be authenticated, use
> skip_before_action with the only: ... option, e.g. in WelcomeController,
> to make the index action available to all visitors, including anonymous
> users:
>
> class WelcomeController < ApplicationController
>   skip_before_action :authenticate_user!, only: :index
>
>   def index
>   end
> end
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HelenaWu/blocitoff/pull/5/files#r42251981>.
>